### PR TITLE
Expand dimensionality notebook

### PR DESCRIPTION
**Pull request by ricardoV94**
I thought it would be useful to expand the dimensionality notebook. It might be a bit too long detailed right now, but I is is probably is easier to discuss from more -> less than the other way around.

I also feel it is better to separate the ellipsis into it's own section and that is not yet finished!

Apologies for tagging so many people for review, but this seems to be a critical part of our documentation that can benefit from varied inputs

Rendered at: https://pymc--5746.org.readthedocs.build/en/5746/learn/core_notebooks/dimensionality.html

### CODE REVIEW

File: docs/source/learn/core_notebooks/dimensionality.ipynb [hunk 1]
```diff
@@ -9,28 +9,29 @@
    },
```
Comment (ricardoV94) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Perhaps it would be useful to mention that batch dimensions always show up to the left <code>[batch2, batch1, support]</code>. Not sure how/where would it be better to mention it...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='35'/>

**Comment by twiecki** [comment]
Please don't force-push as it confuses reviewnb.

**Comment by ricardoV94** [comment]
> Please don't force-push as it confuses reviewnb.

ReviewNB seems useless in this case anyway. I suggest checking the rendered docs directly: https://pymc--5746.org.readthedocs.build/en/5746/learn/core_notebooks/dimensionality.html

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p> </p><ul><li>*Support&nbsp;dimension* → The core&nbsp;dimensionality&nbsp;of a distribution</li></ul><p> </p><p>Maybe add a running example for these, will make it much clearer immediately. Like MvNormal 3x3, batched 4 times. Support=2, batched=4, implied=3...</p><h3><br></h3><h3>"When prototyping implied dimensions are convenient</h3><p> </p><p>"</p><h3>-&gt; When prototyping, implied dimensions are convenient</h3><p> </p><p>I would also mention that this document is more for advanced users. Most users should be fine with just knowing how dims work and use that as best practice. Ideally we'd have a separate beginner tutorial just on that.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>I think more verbose output would make this cleaner with print statements.</p><p>Also, set the numpy default digits to 2.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>hard-coded</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Here I first thought you meant we rarely use&nbsp;<code>size</code>&nbsp;and rather shape, but you mean the stack approach, so should clarify what you mean with first and second.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='15'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>More often&nbsp;<em>than</em>&nbsp;not,</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='55'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>pretty-printing would be nice here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='56'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>It's not that new.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='61'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>black</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='62'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Will those show up in the docs?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='64'/>

Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Maybe remove for now and leave for a separate PR.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='68'/>

**Comment by twiecki** [comment]
It's nice for review though.

**Comment by ricardoV94** [comment]
> It's nice for review though.

You can't see the graphviz outputs. But what did force-push screw up?

**Comment by twiecki** [comment]
> You can't see the graphviz outputs. But what did force-push screw up?

It didn't allow me to post the comments so I had to copy&paste them over.

**Comment by OriolAbril** [comment]
To the danger of becoming a broken record, I think we should define the intended audience of this notebook so that we are all on the same page and can make meaningful reviews.

For example,

> I would also mention that this document is more for advanced users. Most users should be fine with just knowing how dims work and use that as best practice. Ideally we'd have a separate beginner tutorial just on that.

I thought the idea was originally to make a beginner level doc, not really sure now, but I do agree there seems to be a tension here. IMO the scope is quite ambitious and more fitting of advanced level notebook, but the style is more beginner like. i.e. beginners probably don't need to care about all 3 shape, size and dims, 2 at most, and at the same time advanced users don't need to be taught to use size/dims instead of list comprehensions of variables

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (michaelosthege) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Maybe also rephrase:</p><ul><li>*Support dimensions* → Dimensions that can never be taken away from the distribution.</li></ul>

Comment (lucianopaz) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>I haven’t finished the full notebook yet so maybe you talk about this later, but maybe we could also mention that the support dimensions are consumed by the elementwise  logp</p>

Comment (lucianopaz) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Totally agree, you could add it here, when you show what happens with size. Or with shape using ellipsis.</p>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>You can't be explicit with <code>shape</code>. If you try this:</p><p><code>pm.MvNormal.dist(mu=np.zeros(5), cov=np.eye(5), shape=(3,)).eval()</code> you get back an array length 5. It doesn't error out because shape is incompatible with the implied support. The last entry of shape is effectively ignored. This is the case even when <code>mu</code> has shape <code>(1,)</code> :</p><p><code>pm.MvNormal.dist(mu=[0], cov=[[1.]], shape=(3,)).eval().shape</code> return <code>(1,)</code> not <code>(3,)</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='31'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>We could also mention the difference at the <code>logp</code> level </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>As with the <code>MvNormal</code>,the <code>(3,)</code> in the shape is ignored:</p><p> <code>pm.draw(pm.Multinomial.dist(n=5, p=[0.1, 0.3, 0.6], shape=(5, 100))).shape</code> returns <code>(5,3)</code>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>There's a typo here:</p><blockquote>sampling a matrix of from 5 independent</blockquote><p>It should read "sampling a matrix of 5 independent...</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>I feel that the concept of <strong>core case</strong> came out of nowhere. Maybe we could introduce it more explicitly. Something like:</p><p> </p><blockquote>To understand what is going on, we need to introduce the concept of parameter core dimensions. The core dimensions of a distribution's parameter are the minimum number of dimensions the parameters need to have, in order to make the distribution be well defined. In the Multinomial distribution, <code>n</code> must at least be an scalar integer, but <code>p</code> must be at least a vector that represents the probability of having an outcome on each category. So, for the Multinomial distribution, <code>n</code> has 0 core dimensions, and <code>p</code> has 1 core dimension.  </blockquote><p>You can then continue from "So if we have a vector..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Does this actually render in an online notebook? I only see <code>&lt;graphviz.dot.Digraph at some_address&gt;</code> </p><p>Maybe we need use <code>view</code> and get a base64 binary representation of the png? If this renders fine once compiled into the docs, forget what I said</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='58'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>This example is <strong>super tricky</strong>! It shows that the dimension names get attached to a certain shape ("colors"-&gt;2, "group"-&gt;4), and then a previously defined dimension can be used on a completely independent RV, in order to make that parameter have the desired batch dimensions!</p><p>we need to talk about this in more detail</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='62'/>

Comment (lucianopaz) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>One thing that we didn't mention is what coords are used in arviz when the model only defines the dimension name and lets the explicit or implicit size/shape determine the final shape of the RV.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Yeah it renders, there's a link to the rendered docs in the top message of the PR</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Yeah, that's not actually what I wanted to do, I wanted to pass the hyperprior as an input to prior. In any case the <code>dims</code> section definitely needs some rework, that was not where I focused most of my attention.</p><p> </p>

**Comment by canyon289** [comment]
@ricardoV94 do you need a review from me before the discussion in https://github.com/pymc-devs/pymc/issues/5754 is settled or are you good? Want to make sure youre supported :)

**Comment by ricardoV94** [comment]
> @ricardoV94 do you need a review from me before the discussion in #5754 is settled or are you good? Want to make sure youre supported :)

Good for now! Thanks :D

**Comment by ricardoV94** [comment]
Ready for re-review. I have addressed several comments and removed size explanations for now.

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (twiecki) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>still typo</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Thanks</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Emphasized which one was it</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Added a warning about this</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Added in a couple of places</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Added a warning</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Fixed in a subsequent commit</p>

**Comment by twiecki** [comment]
I started but it looked like several comments were not addressed (like typos).

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Added in subsequent commit</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Changed</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Pre-commit should fail, no?</p><p> </p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Made the example less tricky</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>I think it's fine to omit that detail from here, as this is not so concerned with the InferenceData machinery</p>

**Comment by ricardoV94** [comment]
> I started but it looked like several comments were not addressed (like typos).

Yeah I forgot to go over the typos, will push again in a sec

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (twiecki) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>We don't run it for NBs. In general there are a few pep8 violations.</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: Why don't we run it?

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: Done

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: I would not do it here

**Comment by ricardoV94** [comment]
Okay, I cleaned and force-pushed for the last time for a while. Check the second commit in ReviewNB to see the differences between now and the original PR commit

**Comment by ricardoV94** [comment]
Regarding black, the only sensible thing is to add it to the pre-commit. Why haven't we done so already?

**Comment by ricardoV94** [comment]
ReadTheDocs is erroring out with
```
Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/pymc/conda/5746/lib/python3.8/site-packages/jupyter_client/kernelspec.py", line 294, in get_kernel_spec
    raise NoSuchKernel(kernel_name)
jupyter_client.kernelspec.NoSuchKernel: No such kernel named pymc
```

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: Might have missed this typo

**Comment by michaelosthege** [comment]
@ricardoV94 check the source of the `*.ipynb` at the JSON level. These notebooks sometimes (ß?!) save which kernel they were executed with.

**Comment by ricardoV94** [comment]
> @ricardoV94 check the source of the `*.ipynb` at the JSON level. These notebooks sometimes (ß?!) save which kernel they were executed with.

Yeah probably something like that

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (twiecki) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Just an oversight I think because we don't have many NBs here.</p>

File: docs/source/learn/core_notebooks/dimensionality.ipynb [hunk 2]
```diff
@@ -9,231 +9,1402 @@
...
```
Comment (twiecki) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>This is all heading.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='49'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Are coords not explicit?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>broadcasted</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='20'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>this sort of error</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='22'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Nice example!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='29'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>"In the MvNormal examples we just saw," This is a little confusing because its not the example we just saw, but the one two back</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='44'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Showing the actual draws here in addition to the shape would be better. I got very confuysed because in the last examples we show the actual draws, but then reading we switched to shape so was wondering why there wasnt an array</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='45'/>

Comment (canyon289) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Theres all these new words introduced like implcit dimensions, or parameter core dimensions, but theyre not listed at the top. I think its good if they were so its clear what is going to be explained throughout this doc, and provides an easy reference. If they can be linked to the heading in this notebook thatd be even better</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>It's referring to the last two examples with the MvNormal. </p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Sure</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Yeah, I am not sure why...</p>

**Review by OriolAbril — COMMENTED** [review]
RE rtd failure: we need to do a couple updates to `conf.py`

https://github.com/ricardoV94/pymc/blob/0e21bb7e928aeab01e8a875ae46616a0650f3beb/docs/source/conf.py#L147-L154

Should be updated to:

```python
# myst config
nb_execution_mode = "force"
nb_kernel_rgx_aliases = {".*": "python3"}
myst_enable_extensions = ["colon_fence", "deflist", "dollarmath", "amsmath", "substitution"]
myst_substitutions = {
    "version_slug": rtd_version,
}
myst_heading_anchors = None
```

The failure is due to the `nb_kernel...` one. By default, sphinx executes each notebook with the kernel name listed in the ipynb file which is pymc here, not python3 like the other notebooks. Adding this overrides the kernel provided by the notebook and executes all notebooks with the python3 kernel (the only one we have installed on readthedocs). The other changes are updating/removing deprecated parameters.

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 2 above)
Comment (OriolAbril) [inline docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>The initial underscore is part of the target definition syntax in rst. The referenced target should be <code>numpy:basics.broadcasting</code>.  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc/pull/5746/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='22'/>

Comment (canyon289) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: Explicitly stating last two examples would be nice then!

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Coords are explicit, but when you define a distribution you used dims. Or what did you mean?</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>I think that all the important terms are mentioned in the beginning, and this text clearly states we need to introduce a  new concept (core dimensions), just to understand this example. I don't think this concept is as critical to be listed at the top.</p>

Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Thanks!</p>

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 1 above)
Comment (ricardoV94) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Black is running now!</p>

**Comment by ricardoV94** [comment]
Ready for re-review

**Comment by OriolAbril** [comment]
The text in all admonitions ends with `:::`: https://pymcio--5746.org.readthedocs.build/projects/docs/en/5746/learn/core_notebooks/dimensionality.html. Not sure why, but probably related to breaklines. I see the last command removes the ending colon fence which might end up in strange rendering issues at some point.

In admonitions, the text can start in the same line as the opening of the directive, but the closing fence must always be on a different line. The line in which the content is written defines how sphinx interprets that content, see https://sphinx-primer.readthedocs.io/en/latest/core/directives.html

**Comment by ricardoV94** [comment]
> ns, the text can start in the same line as the opening of the directive, but the closing fence must always be on a different line. The line in which the content is written defines how

Fixed it

File: docs/source/learn/core_notebooks/dimensionality.ipynb (hunk 2 above)
Comment (twiecki) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: <p>Don't you use a <code>##</code> thing? </p>

Comment (OriolAbril) [reply docs/source/learn/core_notebooks/dimensionality.ipynb:9]: This is reviewnb messing up, the rendered version looks right: https://pymcio--5746.org.readthedocs.build/projects/docs/en/5746/learn/core_notebooks/dimensionality.html#id1

**Review by twiecki — APPROVED** [review]
